lib/inspec/profile.rb

```diff
@@ -180,6 +181,26 @@ def failed?
       @state == :failed
     end
 
+    def verify_if_signed
+      if signed?
+        verify_signed_profile
+        true
+      else
+        # Log information of unsigned profile
+        Inspec::Log.error "Signature required for profile: #{name}"
+        false
+      end
+    end
+
+    def signed?
+      # Signed profiles have .iaf extension
+      (@source_reader.target.parent.class == Inspec::IafProvider)
+    end
+
+    def verify_signed_profile
+      InspecPlugins::Sign::Base.profile_verify(target, true)
```

Comment:
You may need to ensure the plugin has been loaded at this stage. When run from non-CLI -drivenenvironments - such as kitchen-inspec or compliance-phase - InSpec does not load all Command plugins by default. So this class may not be available.

Reply:
Will test this out